src/shm/pool/format.rs

```diff
@@ -0,0 +1,117 @@
+//! Constants of bytes for pixel associated to wl_buffer's [`Format`]s.
+
+use wayland_client::protocol::wl_shm::Format;
+
+pub fn bytes_per_pixel(format: &Format) -> u32 {
```

Comment:
Not immediately relevant for this pull request, this seems like data we could have in a crate that smithay (https://github.com/Smithay/smithay/pull/587) and sctk both could share.

Since drm formats are effectively the same as the wayland formats we could just have the macro generate functions for wl shm formats.